src/Sylius/Component/Grid/Filter/EntityFilter.php

```diff
@@ -29,8 +29,16 @@ public function apply(DataSourceInterface $dataSource, $name, $data, array $opti
         }
 
         $field = isset($options['field']) ? $options['field'] : $name;
+        $fields = isset($options['fields']) ? $options['fields'] : [$field];
```

Comment:
Anyway, even if `field` option is a more common use-case, replacing it with `fields` is pretty straightforward, we just change:

```yaml
sylius_grid:
  my_grid:
    # ...
    filters:
      field:
        type: entity
        options:
          field: field
```

Into:

```yaml
sylius_grid:
  my_grid:
    # ...
    filters:
      field:
        type: entity
        options:
          fields: [field]
```

Three characters more in the config, but lots of nasty logic removed from code.